### PR TITLE
Refactor Reconciliation Mail Service to generate and email reports manually

### PR DESCRIPTION
**Pull request by aliveni**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-1399

### Change description ###
Duplicate of #707 
Add `automatedEmail` boolean flag to ReconcilitionMailService methods to skip the report already sent condition.
When manually triggered to generate and send reports and the automated email was sent already, it should generate and email the reports again.
This will be used in the next PR.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/blobrouter/reconciliation/service/ReconciliationMailService.java [hunk 1]
```diff
@@ -81,25 +81,27 @@ public void process(LocalDate date, List<TargetStorageAccount> availableAccounts
...
         ReconciliationReport reconciliationReport,
-        TargetStorageAccount account
+        TargetStorageAccount account,
+        boolean automatedEmail
     ) {
         try {
             if (reconciliationReport.summaryContent != null) {
 
-                if (reconciliationReport.sentAt != null) {
+                if (automatedEmail && reconciliationReport.sentAt != null) {
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/blobrouter/reconciliation/service/ReconciliationMailService.java:104]: so it stops here never sends an email if it's automated?

**Comment by rkondratowicz** [comment]
Will this actually be needed if a new report gets generated?
This change makes it possible to force resending the same report again (which I think is not what we need)

**Comment by aliveni** [comment]
Closing the PR as the build is failing.

**Comment by aliveni** [comment]
> Will this actually be needed if a new report gets generated?
> This change makes it possible to force resending the same report again (which I think is not what we need)

As discussed, this PR is not required. We send the email only when there is a new report generated.

**Pull request closed without merging**